In [2]:
import numpy as np
import pandas as pd

In [3]:
df=pd.read_csv("EXCEL DATASET.csv")

In [4]:
df.head(4)

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19


In [4]:
df.shape

(1200, 14)

In [5]:
df.columns

Index(['OrderID', 'Date', 'CustomerID', 'Product', 'Quantity', 'UnitPrice',
       'ShippingAddress', 'PaymentMethod', 'OrderStatus', 'TrackingNumber',
       'ItemsInCart', 'CouponCode', 'ReferralSource', 'TotalPrice'],
      dtype='object')

#### Dataset Column Description

This dataset contains information about customer orders placed in an e-commerce platform. Below is a brief description of each column.

**OrderID** - Unique identifier assigned to each customer order. 

**Date** - The date on which the order was placed. 

**CustomerID** - Unique identifier for each customer.

**Product** - Name of the product purchased by the customer. 

**Quantity** - Number of units of the product ordered.

**UnitPrice** - Price of a single unit of the product.

**ShippingAddress** - Delivery address where the order is shipped.

**PaymentMethod** - Payment mode used by the customer (e.g., Credit Card, UPI, PayPal). 

**OrderStatus** - Current status of the order (e.g., Delivered, Shipped, Pending, Cancelled).

**TrackingNumber** - Unique tracking number used to monitor shipment status. 

**ItemsInCart** - Total number of items present in the customer's cart at the time of purchase.

**CouponCode** - Promotional or discount coupon applied to the order, if any. 

**ReferralSource** - Source through which the customer reached the platform (e.g., Social Media, Search Engine, Direct). 

**TotalPrice** - Total amount paid for the order after considering quantity and any applicable discounts. 

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1200 entries, 0 to 1199
Data columns (total 14 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   OrderID          1200 non-null   object 
 1   Date             1200 non-null   object 
 2   CustomerID       1200 non-null   object 
 3   Product          1200 non-null   object 
 4   Quantity         1200 non-null   int64  
 5   UnitPrice        1200 non-null   float64
 6   ShippingAddress  1200 non-null   object 
 7   PaymentMethod    1200 non-null   object 
 8   OrderStatus      1200 non-null   object 
 9   TrackingNumber   1200 non-null   object 
 10  ItemsInCart      1200 non-null   int64  
 11  CouponCode       891 non-null    object 
 12  ReferralSource   1200 non-null   object 
 13  TotalPrice       1200 non-null   float64
dtypes: float64(2), int64(2), object(10)
memory usage: 131.4+ KB


#### Dataset Contains 1200 rows and 14 columns

In [7]:
df.isnull().sum()

OrderID              0
Date                 0
CustomerID           0
Product              0
Quantity             0
UnitPrice            0
ShippingAddress      0
PaymentMethod        0
OrderStatus          0
TrackingNumber       0
ItemsInCart          0
CouponCode         309
ReferralSource       0
TotalPrice           0
dtype: int64

In [8]:
df["CouponCode"] = df["CouponCode"].fillna("No Coupon")

#### CouponCode contains 309 missing values. Since not every customer uses a coupon, these missing values are valid and were replaced with "No Coupon" instead of deleting the rows.

In [9]:
df.isnull().sum()

OrderID            0
Date               0
CustomerID         0
Product            0
Quantity           0
UnitPrice          0
ShippingAddress    0
PaymentMethod      0
OrderStatus        0
TrackingNumber     0
ItemsInCart        0
CouponCode         0
ReferralSource     0
TotalPrice         0
dtype: int64

In [10]:
df.duplicated().sum()

np.int64(0)

#### There are no duplicate rows.

In [11]:
df.dtypes

OrderID             object
Date                object
CustomerID          object
Product             object
Quantity             int64
UnitPrice          float64
ShippingAddress     object
PaymentMethod       object
OrderStatus         object
TrackingNumber      object
ItemsInCart          int64
CouponCode          object
ReferralSource      object
TotalPrice         float64
dtype: object

In [12]:
df["Date"] = pd.to_datetime(df["Date"])

In [13]:
df.dtypes

OrderID                    object
Date               datetime64[ns]
CustomerID                 object
Product                    object
Quantity                    int64
UnitPrice                 float64
ShippingAddress            object
PaymentMethod              object
OrderStatus                object
TrackingNumber             object
ItemsInCart                 int64
CouponCode                 object
ReferralSource             object
TotalPrice                float64
dtype: object

#### The Date column was initially stored as an object (string) data type. It was converted to the datetime format using `pd.to_datetime()` to ensure accurate date handling, enable time-based analysis, and support operations such as sorting, filtering, and extracting date components (year, month, day).

In [14]:
print("Quantity <= 0 :", (df["Quantity"] <= 0).sum())
print("UnitPrice <= 0 :", (df["UnitPrice"] <= 0).sum())
print("ItemsInCart <= 0 :", (df["ItemsInCart"] <= 0).sum())
print("TotalPrice <= 0 :", (df["TotalPrice"] <= 0).sum())

Quantity <= 0 : 0
UnitPrice <= 0 : 0
ItemsInCart <= 0 : 0
TotalPrice <= 0 : 0


#### There are no invalid numeric values.

In [15]:
df.to_csv("Cleaned_Dataset.csv", index=False)

## Data Cleaning Summary

The following data cleaning steps were performed to prepare the dataset for analysis:

- Identified missing values using `isnull().sum()`.
- Replaced missing values in the **CouponCode** column with **"No Coupon"**, as not all customers use discount coupons.
- Checked for duplicate records using `duplicated()` and found **no duplicate rows** in the dataset.
- Converted the **Date** column from **object** to **datetime** format to enable accurate date-based analysis.
- Verified numeric columns (**Quantity, UnitPrice, ItemsInCart, and TotalPrice**) and confirmed that no invalid (zero or negative) values were present.
- The dataset is now clean, consistent, and ready for exploratory data analysis (EDA) and visualization.

In [16]:
df.describe()

,Date,Quantity,UnitPrice,ItemsInCart,TotalPrice
count,1200,1200.000000,1200.000000,1200.000000,1200.000000
mean,2024-03-22 16:58:48,2.945833,356.412750,5.485000,1053.968300
min,2023-01-01 00:00:00,1.000000,11.390000,1.000000,11.390000
25%,2023-08-03 18:00:00,2.000000,186.062500,4.000000,410.520000
50%,2024-03-23 00:00:00,3.000000,364.210000,5.000000,823.615000
75%,2024-11-08 12:00:00,4.000000,521.570000,7.000000,1578.475000
max,2025-06-30 00:00:00,5.000000,699.930000,10.000000,3456.400000
std,NaN,1.407557,197.177146,2.281983,819.856558


## Summary Statistics Insights

The descriptive statistics provide a comprehensive overview of the numerical features in the dataset.

### Key Observations

- The dataset contains **1,200 complete records**, indicating there are no missing values in the numerical columns.
- Customers purchased an average of **2.95 units** per order, with most orders containing **2 to 4 items**.
- The **average unit price** of products is **356.41**, with prices ranging from **11.39** to **699.93**, showing a diverse product catalog.
- The **average number of items in the cart** is **5.49**, suggesting that customers often add multiple products before completing a purchase.
- The **average order value (TotalPrice)** is **1,053.97**, while the highest recorded order value is **3,456.40**, indicating the presence of some high-value transactions.
- The **standard deviation** of **TotalPrice (819.86)** and **UnitPrice (197.18)** indicates noticeable variation in customer spending and product prices.
- The order dates span from **January 2023 to June 2025**, providing sufficient historical data for trend and time-series analysis.

### Conclusion

Overall, the dataset appears complete and well-structured, with a wide range of product prices and order values. These characteristics make it suitable for performing further exploratory data analysis, identifying purchasing patterns, and generating business insights.